test/test_magnet.cpp

```diff
@@ -476,3 +476,82 @@ TORRENT_TEST(invalid_web_seed_escaping)
 	TEST_CHECK(ec);
 }
 
+TORRENT_TEST(parse_magnet_select_only)
+{
+	error_code ec;
+	add_torrent_params p = parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd"
+		"&dn=foo&so=0,2,4,6-8", ec);
+	TEST_CHECK(!ec);
+
+	TEST_EQUAL(p.file_priorities.size(), 9);
+	for (std::size_t i = 0; i < p.file_priorities.size(); i++)
```

Comment:
I think this would be clearer by doing:
```
auto const yes = default_priority;
auto const no = dont_download;
TEST_CHECK(p.file_priorities == std::vector<download_priority_t>{
   yes, no, yes, no, yes, no, yes, yes, yes});
```
what do you think?

Reply:
yes, better :)